### PR TITLE
feat: initialise proteus with core crypto [FS-1096]

### PR DESCRIPTION
**Pull request by PatrykBuniX**
Use core crypto instead of cryptobox for proteus (and mls) client initialisation.

### CODE REVIEW

File: packages/core/src/Account.ts [hunk 1]
```diff
@@ -356,12 +356,20 @@ export class Account<T = any> extends EventEmitter {
...
+    if (!this.service || !this.apiClient.context) {
       throw new Error('Services are not set.');
     }
 
     try {
-      const localClient = await this.loadAndValidateLocalClient(entropyData);
+      const localClient = await this.loadAndValidateLocalClient();
+      const coreCryptoClient = await this.initCoreCrypto(this.apiClient.context, entropyData);
+
+      await coreCryptoClient.proteusInit();
```
Comment (PatrykBuniX) [inline packages/core/src/Account.ts:367]: do we want to always call `.proteusInit()` even for mls devices? or we should add an if statement here?

Comment (atomrc) [reply packages/core/src/Account.ts:367]: I believe proteus is always going to be init (at least for now). 
MLS should be called on top if MLS is request by the consumer. 

Later (a lot later I believe) we will also condition proteus init, but it's not for now :)

File: packages/core/src/Account.ts [hunk 2]
```diff
@@ -480,23 +512,14 @@ export class Account<T = any> extends EventEmitter {
...
-        this.apiClient.context!,
-        this.cryptoProtocolConfig,
-        entropyData,
-      );
-    }
-
     return loadedClient;
   }
 
+  /*
```
Comment (atomrc) [inline packages/core/src/Account.ts:522]: I guess we can completely delete that function now

File: packages/core/src/client/ClientService.ts [hunk 3]
```diff
@@ -115,16 +116,15 @@ export class ClientService {
...
     const prekeys: PreKey[] = [];
+
     for (let i = 0; i < nbPrekeys; i++) {
       const id = i;
       const key = await coreCryptoClient.proteusNewPrekey(i);
       prekeys.push({id, key: Encoder.toBase64(key).asString});
     }
-    const lastPrekeyId = 65535;
+
+    const lastPrekeyId = ProteusKeys.PreKey.MAX_PREKEY_ID;
```
Comment (PatrykBuniX) [inline packages/core/src/client/ClientService.ts:127]: It stores exact same number

Comment (atomrc) [reply packages/core/src/client/ClientService.ts:127]: Oh perfect, nice catch 👌

**Review by atomrc — APPROVED** [review]

**Pull request merged**